## 生物統計與流行病學 作業四

B10401019 呂祐寬

#### environment:
* Ubuntu 22.04 @ WSL2
* R 4.1.2
* Jupyter Notebook 4.1.3,

In [1]:
R.version.string

[1] "R version 4.1.2 (2021-11-01)"

比較4組獨立樣本，每組樣本數分別為9, 10, 12,與15，以變異數分析(ANOVA)進行顯著水準α=0.05的雙尾檢定。在此，$μ_1$, $μ_2$, $μ_3$, $μ_4$ 分別為第1,2,3,4組的(群體)平均值，此檢定統計量記為“$F^*$ ”，其觀察值記為“ $F_{obs}$”。 

Four groups of independent samples were compared, the number of samples in each group was 9, 10, 12, and 15, respectively, and a two-tailed test with a significance level of α=0.05 was carried out by analysis of variance (ANOVA). Here, $μ_1$, $μ_2$, $μ_3$, $μ_4$ are the (population) mean values of groups 1, 2, 3, and 4 respectively, and this test statistic is marked as "$F^*$", and its observed value is marked as "$F_{obs}$".


<font color="#fcba03">

Q1. (10分)請用給定的數學標記說明ANOVA的虛無與對立假說 <br>
Q1. (10 pts) Please use the given notation to explain the null and alternative hypothesis of ANOVA

</font>

<font color="#039dfc">

### Ans: 
$H_0$: The population means of the four groups are all the same. ($\mu_1=\mu_2=\mu_3=\mu_4$)<br>
$H_1$: The population means of the four groups are not all the same. ($\exists i, j: (i ≠ j) ∧ (μ_i ≠ μ_j)$, where $i, j ∈ {1, 2, 3, 4}$)

</font>


<font color="#fcba03">

. (10分)當虛無假說為真時，$F^*$的(機率)分布的兩個關鍵參數的數值為何? <br>
. (10 pts) What are the values of the two key parameters of the (probability) distribution of $F^*$ when the null hypothesis is true?

</font>
<font color="#039dfc">

### Ans: 
The two parameters of the F distribution in anova are $df_{between}$ and $df_{within}$, which can be calculated as following:
$$df_{between} = k - 1 = 4 - 1 = 3 $$
$$df_{within} = n - k = (9+10+12+15) - 4 = 42$$

</font>

<font color="#fcba03">

Q3. (10 分)接著進行多重比較(multiple comparisons)，說明 Bonferroni 方法如何確保整體顯著水準或整體型 I 錯誤機率不超過 α=5%? <br>
Q3. (10 pts) Then, perform multiple comparisons to explain how the Bonferroni method ensures that the overall significance level or overall type I error rate does not exceed α=5%?

</font>
<font color="#039dfc">

### Ans: 
The overall type I error(Family-wise Error Rate, $FWER$) can be calculated by the individual type I errors between groups:
$$  FWER = \bigcup_{i=1}^{C^k_2}\mathbb{P}(p\leq\alpha') $$
where $\alpha'$ is the type I error of individual type I error between any two groups

The Bonferroni method sets the $\alpha'$ as $\frac{\alpha}{C^k_2}$ , which make $FWER \leq \alpha$ , which can be proved below:

By **Boole's inequality** 
$$  FWER = \bigcup_{i=1}^{C^k_2}\mathbb{P}(p\leq\alpha') \leq \sum_{i=1}^{C^k_2}\mathbb{P}(p\leq\alpha') = \frac{\mathbb{P}(p\leq\alpha)}{C^k_2} \times C^k_2 = \frac{\alpha}{C^k_2} \times {C^k_2} = \alpha$$
$$\Rightarrow FWER \leq \alpha $$

In this case, by setting $\alpha' = \frac{0.05}{C^k_2} = 0.0833$, we can ensure the overall type I error $\alpha\leq0.05$
</font>

<font color="#fcba03">

Q4. (30 分)若 $\bar{X_1}$=170, $\bar{X_2}$=165, $\bar{X_3}$=178, $\bar{X_4}$=185,，皆為身高平均。1,2,3,4 組 分別為某非洲國家東、西、南、北區域的男性抽樣群組。1,2,3,4 組的 身高標準差分別為 11.5, 15.8, 18.6, 12.2，請依據這些數據完成 ANOVA table。 <br>
Q4. (30 pts) If $\bar{X_1}$=170, $\bar{X_2}$=165, $\bar{X_3}$=178, $\bar{X_4}$=185, all are the average height. Groups 1, 2, 3, and 4 are male sample groups in the east, west, south, and north regions of an African country, respectively. The standard deviations of the heights of groups 1, 2, 3, and 4 are 11.5, 15.8, 18.6, and 12.2, respectively. Please complete the ANOVA table based on these data.


</font>

<font color="#039dfc">

### Ans: 
R code provided below

</font>


In [2]:
# https://stackoverflow.com/questions/26170002/anova-in-r-using-summary-data

# package "rpsychi" is archived, 
# and can be downloaded at https://cran.r-project.org/src/contrib/Archive/rpsychi/

# install.packages("gtools")
# install.packages("https://cran.r-project.org/src/contrib/Archive/rpsychi/rpsychi_0.8.tar.gz". repos=NULL)

library(rpsychi)

data.n <- c(9, 10, 12, 15)
data.mean <- c(170, 165, 178, 185)
data.sd <- c(11.5, 15.8, 18.6, 12.2)
ind.oneway.second(data.mean,data.sd,data.n, sig.level=0.05/6)$anova.table # sig level is adjusted using bonferonni

Loading required package: gtools



SS        df MS      F    
Between (A)  2796.457  3 932.152 4.258
Within       9194.080 42 218.907    NA
Total       11990.537 45      NA    NA

In [3]:
pf(4.258, 3, 42, lower.tail = FALSE)

[1] 0.01029728

<font color="#039dfc">

from the reuslts above, we can construct the table below

|        |  SS      |   df |  MS   |   F   |  p   |
|--------|----------|------|-------|-------|------|
|Between |2796.457  |    3 |932.152|4.258  |0.0103|
|Within  |9194.080  |   42 |218.907|       |      |
|Total   |11990.537 |   45 |       |       |      |

</font>

<font color="#fcba03">

Q5.	(10分)承第4題，請根據ANOVA table做出統計判斷，並解釋此統計結果。<br>
Q5.	(10 pts) Following question 4, please make statistical judgments based on the ANOVA table and explain the statistical results.

</font>

<font color="#039dfc">

### Ans: 
With $ p = 0.0103 < 0.05 $, we reject the null hypothesis.

</font>

<font color="#fcba03">

Q6.	(30 分)承第 5 題，請判斷是否需要進行事後比較，若需要，請依照Bonferroni 方法進行統計計算，並判斷是哪一些組別存在統計顯著差異。(用手算請提供完整計算過程，用 R 程式請提供可驗算的 code 並陳述結果) <br>
Q6. (30 pts) Continuing from question 5, please determine whether post-hoc comparison is necessary, and if necessary, perform statistical calculations according to the Bonferroni method, and determine which groups have
statistically significant differences. (If you calculate by hand, please provide the complete calculation process, and if you use R program, please provide the code that can be checked and state the result)

</font>

<font color="#039dfc">

### Ans: 
Since we have rejected the null hypothesis, post-hoc comparison is required to reduce type I error.

In this case, we use Bonferroni method to achieve this using the following code:

</font>

In [4]:
ind.oneway.second(data.mean,data.sd,data.n, sig.level=0.05/6)$raw.contrasts

mean.diff lower   upper  std  
1-2   5       -13.824 23.824 6.798
1-3  -8       -26.066 10.066 6.524
1-4 -15       -32.274  2.274 6.238
2-3 -13       -30.542  4.542 6.335
2-4 -20       -36.726 -3.274 6.040
3-4  -7       -22.867  8.867 5.730

<font color="#039dfc">

From the table above, we can discover that only the confidence interval of 2-4 does not contain 0, which means we can only reject the null hypothesis between group 2 and group 4. <br>
This result tells us that there is significant difference only between groups 2 and 4.

</font>

code provided at: https://github.com/lusteven901228/2023_biostatistics/blob/main/HW4/hw4.ipynb